src/app/containers/ProposalPage/index.tsx

```diff
@@ -1,15 +1,205 @@
-import React from 'react';
+import React, { useState } from 'react';
 
+import dayjs from 'dayjs';
+import duration from 'dayjs/plugin/duration';
 import { Helmet } from 'react-helmet-async';
+import { Box, Flex } from 'rebass/styled-components';
+import styled from 'styled-components';
 
+import { Breadcrumb } from 'app/components/Breadcrumb';
+import { Button, AlertButton } from 'app/components/Button';
 import { DefaultLayout } from 'app/components/Layout';
+import { ProposalModal } from 'app/components/ProposalModal';
+import { Typography } from 'app/theme';
+import { ReactComponent as CalendarIcon } from 'assets/icons/calendar.svg';
+import { ReactComponent as ExternalIcon } from 'assets/icons/external.svg';
+import { ReactComponent as PieChartIcon } from 'assets/icons/pie-chart.svg';
+import { ReactComponent as UserIcon } from 'assets/icons/users.svg';
+
+dayjs.extend(duration);
+const title = 'Distribute more BALN to the DAO fund by reducing the BALN for loans and liquidity pools.';
+const content = `
+Too much income is being given back to people who use Balanced. While incentivization is good, we need to prevent Balanced from becoming a platform that people use purely to earn rewards.\nTo make sure Balanced has enough income in its treasury to cover ongoing costs, like security audits, a higher bug bounty, marketing initiatives, projects that utilize Balanced’s functionality, and so on, we should redirect some of the BALN allocated to borrowers and liquidity providers to the DAO fund instead.
+`;
+
+const metadata = {
+  voted: 16,
+  voters: 748,
+  approvePercentage: 53,
+  rejectPercentage: 14,
+  timestamp: 284400000,
+};
+
+const balnWeight = 2500.12345;
+
+const data = {
+  title,
+  content,
+  metadata,
+  balnWeight,
+};
+
+const ProposalContainer = styled(Box)`
+  flex: 1;
+  border-radius: 10px;
+`;
+
+const ProposalPanel = styled(Box)`
+  flex: 1;
+  border-radius: 10px;
+  padding: 35px 35px;
+  background-color: ${({ theme }) => theme.colors.bg2};
+  margin-bottom: 50px;
+  margin-top: 50px;
+`;
+
+const StatsContainer = styled.span`
+  display: flex;
+  flex-direction: row;
+  flex: 1;
+`;
+
+const Column = styled(Box)`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+`;
+
+const Bar = styled.div`
+  position: relative;
+  height: 15px;
+  width: 100%;
+  background-color: #123955;
+  border-radius: 5px;
+  margin-top: 15px;
+  margin-bottom: 15px;
+`;
+
+const ApproveFill = styled.div<{ percentage: string }>`
+  background: ${({ theme }) => theme.colors.primary};
+  height: 100%;
+  border-top-left-radius: 5px;
+  border-bottom-left-radius: 5px;
+  transition: width 0.2s ease-in;
+  width: ${props => `${props.percentage}%`};
+`;
+
+const RejectFill = styled.div<{ percentage: string }>`
```

Comment:
Could you generalize ApproveFill and RejectFill? I'd like to make one component for ApproveFill and RejectFill.